addons/common/fnc_isTurnedOut.sqf

```diff
@@ -27,54 +27,46 @@ Examples:
 SCRIPT(isTurnedOut);
 
 params ["_unit"];
 
-private _turn = false;
-private _out = false;
+if (isNull objectParent _unit) exitWith {true};
+if (isTurnedOut _unit) exitWith {true};
+
+private _return = false;
 private _vehicle = vehicle _unit;
 private _cfg = configFile >> "CfgVehicles" >> (typeOf _vehicle);
 
-if (_vehicle != _unit) then {
-    if (driver _vehicle == _unit) then {
-        private _forceHideDriver = getNumber(_cfg >> "forceHideDriver");
-        _turn = (_forceHideDriver != 1);
-        _out = (_vehicle animationPhase "hatchDriver") > 0;
-    } else {
-        private _assignedRole = assignedVehicleRole _unit;
-        if (_assignedRole select 0 == "Turret") then {
-            private _turretPath = _assignedRole select 1;
-            private _turret = [_vehicle, _turretPath] call CBA_fnc_getTurret;
-
-            private _canHideGunner = getNumber(_turret >> "canHideGunner");
-            private _forceHideGunner = getNumber(_turret >> "forceHideGunner");
-            _turn = (_canHideGunner == 1 && _forceHideGunner == 0);
-
-            private _hatchAnimation = getText(_turret >> "animationSourceHatch");
-            _out = (_vehicle animationPhase _hatchAnimation) > 0;
-
-        } else {
-            if ((_assignedRole select 0) == "Cargo") then {
-                private _attenuateCargo = getArray(_cfg >> "soundAttenuationCargo");
-                if ((count _attenuateCargo) > 0) then {
-                    private _index = (count _attenuateCargo)-1; // wait for command to get cargo index
+if (driver _vehicle == _unit) then {
+    _return = (_vehicle animationPhase "hatchDriver") > 0;
+} else {
+    (assignedVehicleRole _unit) params [["_position", ""], ["_turretPath", []]];
+    
+    if (_position == "Turret") then {
+        private _turret = [_vehicle, _turretPath] call CBA_fnc_getTurret;
 
-                    if (_index > -1) then {
-                        if(_index > (count _attenuateCargo)-1) then {
-                            _index = (count _attenuateCargo)-1;
-                        };
-                        if((_attenuateCargo select _index) == 0) then {
-                            _out = true;
-                        };
-                    };
-                };
+        private _gunnerAction = getText (_turret >> "gunnerAction");
+        private _hatchAnimation = getText (_turret >> "animationSourceHatch");
+        
+        _return = (_vehicle animationPhase _hatchAnimation) > 0 || animationState player == _gunnerAction;
+    } else {
+        if (_position == "Cargo") then {
+            private _cargoIndex = _vehicle getCargoIndex _unit;
+            private _window = ['ani_window_1', 'ani_window_2', 'ani_window_4', 'ani_window_3'] select (_cargoIndex + 1);
+            if ((_vehicle doorPhase _window) > 0) exitWith {_return = true};
+            
+            if ([_unit] call ace_common_fnc_getTurretIndex in ([_vehicle] call ace_common_fnc_getTurretsFFV)) exitWith {_return = true};
```

Comment:
These are unavailable when ACE is not used. Bad idea.

Reply:
I wrote it as attention in the first message and as @ mention comment to you https://github.com/CBATeam/CBA_A3/pull/1015#issuecomment-431695148

Can i add them to cba... or is there an other alternative?

Reply:
Will thonk about to tomorrow morning.